apis/r/R/SOMASparseNDArray.R

```diff
@@ -227,14 +224,26 @@ SOMASparseNDArray <- R6::R6Class(
       if (!is.null(private$tiledb_timestamp)) {
           arr@timestamp <- private$tiledb_timestamp
       }
-      ## spdl::debug("[SOMASparseNDArray] '.write_coo_dataframe' layout '{}' is_sparse '{}' ",
-      ##             tiledb::query_layout(arr), private$.is_sparse)
-      ## if (!private$.is_sparse && tiledb::query_layout(arr) == "UNORDERED") {
-      ##     tiledb::query_layout(arr) <- "GLOBAL_ORDER"
-      ##     cat("*********", tiledb::query_layout(arr), "*****\n")
-      ##     print(arr)
-      ## }
-      arr[] <- values
+      nms <- colnames(values)
+
+      ## the 'soma_data' data type may not have been cached, and if so we need to fetch it
+      if (is.null(private$.type)) {
+          ## TODO: replace with a libtiledbsoma accessor as discussed
```

Comment:
Are you able to use [SOMAArray::arrow_schema](https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/soma/soma_array.h#L564-L567) or [SOMASparseArrayNDArray::schema](https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/soma/soma_sparse_ndarray.h#L154-L159) to get the dtype of `soma_data`?

Reply:
I think it is not clear where the value would come from.  This function is called with just the values to be written, and this segment is where, if no 'cached' type info exists, we go to the schema on disk as a last resort.

https://github.com/single-cell-data/TileDB-SOMA/blob/f06cd71b3db4cd98d6263013285b8650849a5af2/apis/r/R/SOMASparseNDArray.R#L229-L235

In some of the use case we do not need that as `private$.type` has been set.

Reply:
And it is similar for thw `write()` function from which this is called.  It gets called with just `values` (and an optional bbox).  No schema info.